source/beamline/SGM/monochromator/SGMEnergyTrajectory.cpp

```diff
@@ -0,0 +1,199 @@
+#include "SGMEnergyTrajectory.h"
+
+SGMEnergyTrajectory::SGMEnergyTrajectory(double startEnergy,
+                                         double endEnergy,
+                                         double time,
+                                         SGMGratingSupport::GratingTranslation gratingTranslation)
+{
+
+    startMonoInfo_ = new SGMMonochromatorInfo(startEnergy, gratingTranslation);
+    startEnergy_ = startEnergy;
+    endMonoInfo_ = new SGMMonochromatorInfo(endEnergy, gratingTranslation);
+    endEnergy_ = endEnergy;
+
+    // ensure both undulator harmonics are the same.
+    if(startMonoInfo_->undulatorHarmonic() != endMonoInfo_->undulatorHarmonic()) {
+
+        // If there is a conflict, just use the first harmonic.
+        startMonoInfo_->setAutoDetectUndulatorHarmonic(false);
+        endMonoInfo_->setAutoDetectUndulatorHarmonic(false);
+
+        startMonoInfo_->setUndulatorHarmonic(SGMUndulatorSupport::FirstHarmonic);
+        endMonoInfo_->setUndulatorHarmonic(SGMUndulatorSupport::ThirdHarmonic);
```

Comment:
If we are just using the first harmonic like your previous comment mentions, shouldn't this line also be `SGMUndulatorSupport::FirstHarmonic`?